source/crud/bulk-write.md

```diff
@@ -0,0 +1,901 @@
+`errorsOnly` as `true`.
+
+### `ordered`
+
+The `ordered` field defines whether writes should be executed in the order in which they were
+specified, and, if an error occurs, whether the server should halt execution of further writes. It
+is optional and defaults to true on the server. Drivers MUST explicitly define `ordered` as `true`
+in the `bulkWrite` command if a value is not specified in `BulkWriteOptions`. This is required to
+avoid inconsistencies between server and driver behavior if the server default changes in the
+future.
+
+### Size Limits
+
+The server reports a `maxBsonObjectSize` in its `hello` response. This value defines the maximum
+size for documents that are inserted into the database. Documents that are sent to the server but
+are not intended to be inserted into the database (e.g. command documents) have a size limit of
+`maxBsonObjectSize + 16KiB`. When an acknowledged write concern is used, drivers MUST NOT perform
+any checks related to these size limits and MUST rely on the server to raise an error if a limit is
+exceeded. However, when an unacknowledged write concern is used, drivers MUST raise an error if one
+of the following limits is exceeded:
+
+- The size of a document to be inserted MUST NOT exceed `maxBsonObjectSize`. This applies to the
+  `document` field of an `InsertOneModel` and the `replacement` field of a `ReplaceOneModel`.
+- The size of an entry in the `ops` array MUST NOT exceed `maxBsonObjectSize + 16KiB`.
+- The size of the `bulkWrite` command document MUST NOT exceed `maxBsonObjectSize + 16KiB`.
+
+See [SERVER-10643](https://jira.mongodb.org/browse/SERVER-10643) for more details on these size
+limits.
+
+## Command Batching
+
+Drivers MUST accept an arbitrary number of operations as input to the `MongoClient.bulkWrite`
+method. Because the server imposes restrictions on the size of write operations, this means that a
+single call to `MongoClient.bulkWrite` may require multiple `bulkWrite` commands to be sent to the
+server. Drivers MUST split bulk writes into separate commands when the user's list of operations
+exceeds one or more of these maximums: `maxWriteBatchSize`, `maxBsonObjectSize` (for `OP_MSG`
+payload type 0), and `maxMessageSizeBytes` (for `OP_MSG` payload type 1). Each of these values can
+be retrieved from the selected server's `hello` command response. Drivers MUST merge results from
+multiple batches into a single `BulkWriteResult` or `BulkWriteException` to return from
+`MongoClient.bulkWrite`.
+
+When constructing the `nsInfo` array for a `bulkWrite` batch, drivers MUST only include the
+namespaces that are referenced in the `ops` array for that batch.
+
+### Number of Writes
+
+`maxWriteBatchSize` defines the total number of writes allowed in one command. Drivers MUST split a
+bulk write into multiple commands if the user provides more than `maxWriteBatchSize` operations in
+the argument for `models`.
+
+### Total Message Size
+
+#### Encrypted bulk writes
```

Comment:
Out of caution, I suggest:

- Remove mention of supporting auto-encryption: https://github.com/mongodb/specifications/commit/ed483b5e9aee3f78d333cdbe9d9ea411b3d82770
- Return an error if auto-encryption is enabled: https://github.com/mongodb/specifications/commit/e39c82c4f68786d7dd29c0c1b4e1c342bf3cf069
- Add prose test to verify: https://github.com/mongodb/specifications/commit/62814a38b71d9edd796af9f5f0dacf487ead5e83

Rationale: This PR does not include tests for auto encryption. Tests are planned in DRIVERS-2888. As specified, a driver may implement `MongoClient.bulkWrite` from this PR and bypass use of libmongocrypt. That may result in a `MongoClient` configured with auto encryption sending a `bulkWrite` and ignoring auto encryption settings (may send plaintext when ciphertext was intended).

These changes may be soon-removed once DRIVERS-2888 is resolved. But I prefer to keep this PR self-contained. This may enable a driver to release `MongoClient.bulkWrite` without DRIVERS-2888.